src/imitation/rewards/reward_net.py

```diff
@@ -49,101 +48,144 @@ def __init__(
-        placeholders to be filled to calculate this output, but for all
-        subclasses, filling the following placeholders will be sufficient:
+    def predict_reward_train(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+    ) -> np.ndarray:
+        """Compute the train reward with raw ndarrays, including preprocessing.
 
-        ```
-        self.obs_ph
-        self.act_ph
-        self.next_obs_ph
-        ```
+        Args:
+          state: current state. Leading dimension should be batch size B.
+          action: action associated with `state`.
+          next_state: next state.
+          done: 0/1 value indicating whether episode terminates on transition
+            to `next_state`.
 
         Returns:
-          tf.Tensor: A (None,) shaped Tensor holding
-            the test reward associated with each timestep.
+          np.ndarray: A (B,)-shaped ndarray holding
+              the training reward associated with each timestep.
         """
-        return self._theta_output
+        return self._eval_reward(True, state, action, next_state, done)
 
-    @abstractmethod
-    def build_theta_network(
-        self, inputs: Sequence[tf.Tensor]
-    ) -> Tuple[tf.Tensor, networks.LayersDict]:
-        """Builds the test reward network.
+    def predict_reward_test(
+        self,
+        state: np.ndarray,
+        action: np.ndarray,
+        next_state: np.ndarray,
+        done: np.ndarray,
+    ) -> np.ndarray:
+        """Compute the test reward with raw ndarrays, including preprocessing.
 
-        The output of the network is the same as the reward used for transfer
-        learning, and is the Tensor returned by `self.reward_output_test()`.
+        Note this is the reward we use for transfer learning.
 
         Args:
-          obs_input: The observation input. Its shape is
-              `((None,) + self.env.observation_space.shape)`.
-          act_input: The action input. Its shape is
-              `((None,) + self.env.action_space.shape)`. The None dimension is
-              expected to be the same as None dimension from `obs_input`.
+          state: current state. Lead dimension should be batch size B.
+          action: action associated with `state`.
+          next_state: next state.
+          done: 0/1 value indicating whether episode terminates on transition
+            to `next_state`.
 
         Returns:
-          A tuple (theta_output, layers) where
-            * theta_output is a reward prediction for each of the inputs,
-              of shape `(None,)`;
-            * layers is a dictionary mapping to individual checkpointable layers.
+          np.ndarray: A (B,)-shaped ndarray holding the test reward
+            associated with each timestep.
         """
-        pass
+        return self._eval_reward(False, state, action, next_state, done)
+
+    def _eval_reward(
```

Comment:
This mostly duplicates `DiscrimNet._eval_reward` although fixing this should probably be punted to when we do the `DiscrimNet` vs `RewardNet` refactor.

Reply:
Agreed